src/mpid/ch4/shm/posix/posix_coll_release_gather.h

```diff
@@ -53,7 +53,6 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_bcast_release_gather(void *buffer,
     MPI_Aint lb, true_lb, true_extent, extent, type_size;
     void *ori_buffer = buffer;
     MPI_Datatype ori_datatype = datatype;
-    int cellsize = MPIDI_POSIX_RELEASE_GATHER_BCAST_CELLSIZE;
```

Comment:
Is moving this setting later necessary? Unclear to me how this affects the behavior.

Reply:
It is not "necessary", but for readability. `cellsize` is only used in `MPIR_Algo_calculate_pipeline_chunk_info` and the reason we need this variable is due to the exception under `#ifdef HAVE_ERROR_CHECKING`. So move them closer makes the code easier to read. I can split this change if you prefer.

Reply:
I have split the commits.